File: src/main/java/com/qwwuyu/server/dao/FlagMapper.java
package com.qwwuyu.server.dao;

import com.qwwuyu.server.bean.Flag;
import org.apache.ibatis.annotations.Param;

import java.util.List;

public interface FlagMapper {
    Flag selectByPrimaryKey(Integer id);

    int insert(Flag record);

    int updateByPrimaryKeySelective(Flag record);

    int deleteByPrimaryKey(Integer id);

    List<Flag> selectByFlag(@Param("flag") Flag flag, @Param("limit") int limit, @Param("offset") int offset, @Param("asc") String asc, @Param("desc") String desc);
}

File: app/src/main/kotlin/com/qwwuyu/gs/mapper/FlagMapper.kt
package com.qwwuyu.gs.mapper

import com.qwwuyu.gs.entity.Flag
import org.apache.ibatis.annotations.Mapper
import org.apache.ibatis.annotations.Param

@Mapper
interface FlagMapper {
    fun selectByPrimaryKey(id: Int): Flag?

    fun insert(record: Flag): Int

    fun updateByPrimaryKeySelective(record: Flag): Int

    fun deleteByPrimaryKey(id: Int): Int

    fun selectByFlag(@Param("flag") flag: Flag, @Param("limit") limit: Int, @Param("offset") offset: Int,
                     @Param("asc") asc: String?, @Param("desc") desc: String?): List<Flag>
}